src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -39,46 +162,1551 @@ def points_gridding(self, points, method="linear", coarsen=1):
+
+            # Handle coincident points
+            zero_dist = distances < 1e-10
+            if np.any(zero_dist):
+                zi_flat[i] = z_nearby[zero_dist][0]
+                continue
+
+            # IDW weights
+            weights = 1.0 / (distances**power)
+            zi_flat[i] = np.sum(weights * z_nearby) / np.sum(weights)
+    else:
+        # Simple vectorized approach for small datasets without radius
+        logger.debug(
+            "IDW gridding %d points to %d grid nodes (power=%.2f, using all points)...",
+            npoints,
+            len(xi_flat),
+            power,
+        )
+
+        for i in range(len(xi_flat)):
+            # Calculate distances from grid point to all data points
+            dx = xcv - xi_flat[i]
+            dy = ycv - yi_flat[i]
+            distances = np.sqrt(dx**2 + dy**2)
+
+            z_nearby = zcv
+
+            # Skip if not enough points
+            if len(distances) < min_points:
+                continue
+
+            # Handle coincident points (distance = 0)
+            zero_dist = distances < 1e-10
+            if np.any(zero_dist):
+                zi_flat[i] = z_nearby[zero_dist][0]
+                continue
+
+            # Calculate weights: w_i = 1 / d_i^power
+            weights = 1.0 / (distances**power)
+
+            # Normalize weights and calculate weighted average
+            zi_flat[i] = np.sum(weights * z_nearby) / np.sum(weights)
+
+    # Reshape back to grid
+    znew = zi_flat.reshape(xiv.shape)
+
+    logger.debug("IDW gridding ... DONE")
+
+    self._ensure_correct_values(znew)
+
+
+def _range_to_len_scale(
+    range_value: float | tuple[float, float],
+    variogram_model: str,
+    alpha: float | None = None,
+) -> float | tuple[float, float]:
+    """
+    For kriging. Convert RMS/Petrel-style 'range' to GSTools 'len_scale'.
+
+    Args:
+        range_value: Range parameter (distance to sill). Can be:
+            - float: isotropic range
+            - tuple of 2 floats: anisotropic range_value (range_x, range_y)
+        variogram_model: Model name
+        alpha: Shape parameter for Stable model (1.0-2.0)
+
+    Returns:
+        len_scale for GSTools (float or tuple)
+    """
+    model = variogram_model.lower()
+
+    # Determine conversion factor based on model
+    if model == "gaussian":
+        factor = np.sqrt(3)
+    elif model == "exponential":
+        factor = 3.0
+    elif model == "spherical":
+        factor = 1.0
+    elif model == "stable":
+        if alpha is None:
+            logger.warning("Alpha not provided for Stable model, assuming alpha=1.5")
+            alpha = 1.5
+        # Linear interpolation between Exponential and Gaussian conversions
+        factor = 3.0 - (alpha - 1.0) * (3.0 - np.sqrt(3))
+    elif model == "matern":
+        factor = 3.0  # assuming nu=0.5 (exponential-like)
+    else:
+        logger.warning(f"Unknown conversion for {variogram_model}, using range_value/2")
+        factor = 2.0
```

Comment:
I see `Linear` is documented as an option as `variogram_model` in the `points_kriging_gridding` method, but I don't see `linear` as an option here it will hence get the factor 2. Is this intended?